backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresExtension.java

```diff
@@ -0,0 +1,143 @@
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresExtension.class);
+
+    private static void displayDockerLog(OutputFrame outputFrame) {
+        LOGGER.info(outputFrame.getUtf8String().trim());
+    }
+
+    private final PostgreSQLContainer postgresContainer;
+    private final PostgresModule postgresModule;
+    private PostgresqlConnectionFactory connectionFactory;
+    private PostgresExecutor postgresExecutor;
+    private PostgresTableManager postgresTableManager;
+
+    public PostgresExtension(PostgresModule postgresModule) {
+        this.postgresContainer = new PostgreSQLContainer<>(PostgresFixture.IMAGE)
+            .withDatabaseName(PostgresFixture.Database.DB_NAME)
+            .withUsername(PostgresFixture.Database.DB_USER)
+            .withPassword(PostgresFixture.Database.DB_PASSWORD)
+            .withLogConsumer(PostgresExtension::displayDockerLog)
+            .withCreateContainerCmdModifier(cmd -> cmd.withName("james-postgres-test-" + UUID.randomUUID()));
+
+        this.postgresModule = postgresModule;
+    }
+
+    public PostgresExtension() {
+        this.postgresContainer = new PostgreSQLContainer<>(PostgresFixture.IMAGE)
+            .withDatabaseName(PostgresFixture.Database.DB_NAME)
+            .withUsername(PostgresFixture.Database.DB_USER)
+            .withPassword(PostgresFixture.Database.DB_PASSWORD)
+            .withLogConsumer(PostgresExtension::displayDockerLog)
+            .withCreateContainerCmdModifier(cmd -> cmd.withName("james-postgres-test-" + UUID.randomUUID()));
+
+        this.postgresModule = PostgresModule.EMPTY_MODULE;
+    }
+
+    @Override
+    public void beforeAll(ExtensionContext extensionContext) {
+        if (!postgresContainer.isRunning()) {
+            postgresContainer.start();
+            initPostgresSession();
+        }
+    }
+
+    private void initPostgresSession() {
+        connectionFactory = new PostgresqlConnectionFactory(PostgresqlConnectionConfiguration.builder()
+            .host(getHost())
+            .port(getMappedPort())
+            .username(PostgresFixture.Database.DB_USER)
+            .password(PostgresFixture.Database.DB_PASSWORD)
+            .database(PostgresFixture.Database.DB_NAME)
+            .schema(PostgresFixture.Database.SCHEMA)
+            .build());
+
+        postgresExecutor = new PostgresExecutor(connectionFactory.create()
+            .cast(Connection.class));
+    }
+
+    @Override
+    public void afterAll(ExtensionContext extensionContext) {
+        disposePostgresSession();
+        postgresContainer.stop();
+    }
+
+    private void disposePostgresSession() {
+        postgresExecutor.dispose().block();
+    }
+
+    @Override
+    public void beforeEach(ExtensionContext extensionContext) {
+        initTablesAndIndexes();
+    }
+
+    @Override
+    public void afterEach(ExtensionContext extensionContext) {
+        truncateTablesAndIndexes();
```

Comment:
Should we DROP SCHEMA, and RE-CREATE SCHEMA?

// For table with enable RLS, we need drop POLICY (truncate table and index is not enough for this case)

Code ex: 
```java
Flux.usingWhen(postgresExtension.getConnection()
                    .map(c -> (PostgresqlConnection) c),
                connection -> Mono.from(connection.createStatement("DROP SCHEMA " + PostgresFixture.Database.SCHEMA + " CASCADE").execute())
                    .then(Mono.from(connection.createStatement("CREATE SCHEMA " + PostgresFixture.Database.SCHEMA).execute()))
                    .flatMap(PostgresqlResult::getRowsUpdated),
                Connection::close)
            .collectList()
            .block();
```